File: Java/com.faculdadevincit.designpatterns.facadepattern/src/LojaFacade.java
public class LojaFacade {
    private ProcessadorDePedidos processadorDePedidos;
    private SistemaDePagamentos sistemaDePagamentos;
    private SistemaDeEnvio sistemaDeEnvio;

    public LojaFacade() {
        processadorDePedidos = new ProcessadorDePedidos();
        sistemaDePagamentos = new SistemaDePagamentos();
        sistemaDeEnvio = new SistemaDeEnvio();
    }

    public void fazerPedido(String item, double valor) {
        processadorDePedidos.CriarPedido(item);
        sistemaDePagamentos.ProcessarPagamento(valor);
        sistemaDeEnvio.EnviarPedido(item);

        System.out.println("Pedido realizado com sucesso!");
    }
}

File: CSharp/FaculdadeVincit.DesignPatterns.FacadePattern/LojaFacade.cs
namespace FaculdadeVincit.DesignPatterns.FacadePattern;

public class LojaFacade
{
    private readonly ProcessadorDePedidos _processadorDePedidos;
    private readonly SistemaDePagamentos _sistemaDePagamentos;
    private readonly SistemaDeEnvio _sistemaDeEnvio;

    public LojaFacade()
    {
        _processadorDePedidos = new ProcessadorDePedidos();
        _sistemaDePagamentos = new SistemaDePagamentos();
        _sistemaDeEnvio = new SistemaDeEnvio();
    }

    public void FazerPedido(string item, decimal valor)
    {
        _processadorDePedidos.CriarPedido(item);
        _sistemaDePagamentos.ProcessarPagamento(valor);
        _sistemaDeEnvio.EnviarPedido(item);

        System.Console.WriteLine("Pedido foi realizado com sucesso!");
    }

}
